cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/TopicConfigurationParameters.java

```diff
@@ -30,8 +41,7 @@
  * </pre>
  */
 public class TopicConfigurationParameters extends GoalBasedOptimizationParameters {
-  protected Pattern _topic;
-  protected short _replicationFactor;
+  protected Map<Short, Set<Pattern>> _topicPatternsByReplicationFactor;
```

Comment:
I am worried that this change makes the `TopicConfigurationParameters` strongly tied to the replication factor change request, which is just an instance of topic configuration change request. Is it possible to abstract away the different type topic configuration requests here? -- i.e. `ReplicationFactorParameters` could be used to bundle changes corresponding to replication factor of topics.

Reply:
When I am refactoring the code, I realize that to keep subgroups of parameter inside endpoint's parameter(for `admin` and `topic_configuration` endpoint) become non-trivial so I think it is a good idea to separate the refactor into another patch, I created issue #886 to track it.